pyomo/contrib/mcpp/mcppInterface.cpp

```diff
@@ -11,15 +11,26 @@
 #include "interval.hpp"
 #include "mccormick.hpp"
 #include <sstream>
+#include <string>
 typedef mc::Interval I;
 typedef mc::McCormick<I> MC;
 
 // Module-level variables as utilities to pass information back to Python
 std::string lastException;
 std::string lastDisplay;
+std::string version;
 
 extern "C"
 {
+
+    // Version number
+    const char* get_version() {
+        std::ostringstream vStream;
+        vStream << "19.11.12" << std::flush;
+        version = vStream.str();
+        return version.c_str();
```

Comment:
This is technically a seg fault: you are returning a pointer to an internal data member of an object allocated on the stack.  When the function returns, the `vStream` is deallocated (including the puffer that the `char*` is pointing to).  You are just "lucky" that the OS hasn't reclaimed / reallocated the memory yet.

A safer solution is:
```c++
const char* version_string = "19.11.12";
const char* get_version() {
    return version_string;
}
```
This way, the pointer that is returned is statically allocated as part of the program's global memory space.